src/online/unigrid.jl

```diff
@@ -23,23 +23,76 @@ function inducingpoints(
     Z = map(bounds) do lims
         LinRange(lims..., alg.m)
     end
-    return Z
+    proditer = Iterators.product(Z...)
+    return UniformGrid(proditer)
 end
 
 function updateZ!(
     ::AbstractRNG, Z::AbstractVector, alg::UniGrid, X::AbstractVector; kwargs...
 )
-    ndim = length(Z)
+    ndim = length(X[1])
     new_bounds = [extrema(x -> getindex(x, i), X) for i in 1:ndim]
-    map!(Z, Z, new_bounds) do Z_d, new_b
-        x_start = min(Z_d.start, new_b[1]) # Find the new limits
-        x_stop = max(Z_d.stop, new_b[2])
-        return LinRange(x_start, x_stop, alg.m) # readapt bounds
+    Zn = map(Z.proditer.iterators, new_bounds) do old_Z, new_b
+        x_start = min(old_Z.start, new_b[1])
+        x_stop = max(old_Z.stop, new_b[2])
+        return LinRange(x_start, x_stop, alg.m)
     end
+    Z.proditer = Iterators.product(Zn...)
     return Z
 end
 
-function updateZ(rng::AbstractRNG, Z::AbstractVector, alg::UniGrid, X::AbstractVector)
+function updateZ(
+    rng::AbstractRNG, Z::AbstractVector, alg::UniGrid, X::AbstractVector; kwargs...
+)
     Zn = deepcopy(Z)
-    return updateZ!(rng, Zn, alg, X)
+    return updateZ!(rng, Zn, alg, X; kwargs...)
+end
+
+export UniformGrid
+mutable struct UniformGrid{T,Titer} <: AbstractVector{T}
+    proditer::Titer
+end
+
+function UniformGrid(
+    proditer::Iterators.ProductIterator
+)
+    T = eltype(first(proditer.iterators))
```

Comment:
Ok I went too fast again :sweat_smile:, `T` should be the type of the returned `Vector`s.
So this should be something like
```suggestion
    T = Vector{eltype(first(proditer.iterators))}
```
And again, this can probably improved

Reply:
Looked into a few options, this one is in fact the most efficient one (at least for my small test case, possible that something like `eltype(eltype(...))` becomes more efficient for high-dimensional grids.